### ISSUE TITLE
Need to customise the look and feel of T-Pot

### ISSUE THREAD

**Issue by Blason** [dtag-dev-sec/tpotce#392]
# Issues
Hi There,

Wanted to know if is it possible to change the complete look and feel of T-pot iso? I would like to make the custom ISO with Whitelabling. 
And instead of Debian wondering if Ubuntu or CentOS can be used?
Before you post your issue make sure it has not been answered yet and provide `basic support information` if you come to the conclusion it is a new issue:
- 🔍 Use the [search function](https://github.com/dtag-dev-sec/tpotce/issues?utf8=%E2%9C%93&q=) first
- 🧐 Check our [WIKI](https://github.com/dtag-dev-sec/tpotce/wiki)
- 📚 Consult the documentation of 💻 [Debian](https://www.debian.org/doc/), 🐳 [Docker](https://docs.docker.com/), the 🦌 [ELK stack](https://www.elastic.co/guide/index.html) and the 🍯 [T-Pot Readme](https://github.com/dtag-dev-sec/tpotce/blob/master/README.md).
- **⚠️ Provide [basic support information](#info) or similiar information with regard to your issue or we can not help you and will close the issue without further notice**

<br>
<br>
<br>

<a name="info"></a>
## ⚠️ Basic support information (commands are expected to run as `root`)

- What version of the OS are you currently using `lsb_release -a` and `uname -a`?
- What T-Pot version are you currently using?
- What edition (Standard, Nextgen, etc.) of T-Pot are you running?
- What architecture are you running on (i.e. hardware, cloud, VM, etc.)?
- Did you have any problems during the install? If yes, please attach `/install.log` `/install.err`.
- How long has your installation been running?
- Did you install upgrades, packages or use the update script?
- Did you modify any scripts or configs? If yes, please attach the changes.
- Please provide a screenshot of `glances` and `htop`.
- How much free disk space is available (`df -h`)?
- What is the current container status (`dps.sh`)?
- What is the status of the T-Pot service (`systemctl status tpot`)?
- What ports are being occupied? Stop T-Pot `systemctl stop tpot` and run `netstat -tulpen`
- If a single container shows as `DOWN` you can run `docker logs <container-name>` for the latest log entries

**Comment by Blason** [comment]
> @Blason Are you trying to white-label the entire product suite or just the kibana dashboard?
> 
> As for ubuntu; it can be installed however the latest release will have a number of breaks. If you want ot use it then maybe look at an older version like 17.10.

Well eventually wanted to white-label entire iso to use for the internal purpose that comprises of Kibana as well. At this moment it seems there is no proper documentation as to what each script is doing?

**Comment by iukea1** [comment]
> > @Blason Are you trying to white-label the entire product suite or just the kibana dashboard?
> > As for ubuntu; it can be installed however the latest release will have a number of breaks. If you want ot use it then maybe look at an older version like 17.10.
> 
> Well eventually wanted to white-label entire iso to use for the internal purpose that comprises of Kibana as well. At this moment it seems there is no proper documentation as to what each script is doing?

looks like you got to go digging ;) open source for a reason :D 

The T-Pot guys worked very hard and I don't think they would take two kind for you to white label there product and set it up on some dashboard in a company for shock factor.

**Comment by iukea1** [comment]
@Blason could look at https://github.com/KBDKID/mhn 

It is owned by Alomli and they might come after you for copywriting if you white-label, but I leave that to your company's lawyers.

**Comment by KBDKID** [comment]
@Blason trying to whitelabel the whole iso is going to take longer than just building the honeynet yourself. I am also going to strongly recommend not trying that. Realistically if you wanted to beautiful the dashboard then go for it but trying to rip the entire suite, then best of luck. Modern Honey Net might be something more up your alley as it will teach you how to integrate honeypots into a dashboard of your customization. Threatstream do a great job maintaining it however their deployment scripts certainly need some.looking over. Minor tweaks and changes.

@iukea1 I wouldn't use that git. Working on publishing some revisions for it shortly and it could get very funky.

https://github.com/threatstream/mhn

**Comment by Blason** [comment]
> @Blason trying to whitelabel the whole iso is going to take longer than just building the honeynet yourself. I am also going to strongly recommend not trying that. Realistically if you wanted to beautiful the dashboard then go for it but trying to rip the entire suite, then best of luck. Modern Honey Net might be something more up your alley as it will teach you how to integrate honeypots into a dashboard of your customization. Threatstream do a great job maintaining it however their deployment scripts certainly need some.looking over. Minor tweaks and changes.
> 
> @iukea1 I wouldn't use that git. Working on publishing some revisions for it shortly and it could get very funky.
> 
> https://github.com/threatstream/mhn

Well, even I was thinking that better to build the honeypots by myself rather than picking up the codes from there.

But from a license perspective I believe and feel its a open source product I guess.

**Comment by iukea1** [comment]
> @Blason trying to whitelabel the whole iso is going to take longer than just building the honeynet yourself. I am also going to strongly recommend not trying that. Realistically if you wanted to beautiful the dashboard then go for it but trying to rip the entire suite, then best of luck. Modern Honey Net might be something more up your alley as it will teach you how to integrate honeypots into a dashboard of your customization. Threatstream do a great job maintaining it however their deployment scripts certainly need some.looking over. Minor tweaks and changes.
> 
> @iukea1 I wouldn't use that git. Working on publishing some revisions for it shortly and it could get very funky.
> 
> https://github.com/threatstream/mhn

thanks, bro for the heads up, I heard about that project a while ago and it has just been in my honypots folder on my favorites bar for a while.

Also, I agree with you on it would take longer to white-label the whole iso

**Comment by iukea1** [comment]
> > @Blason trying to whitelabel the whole iso is going to take longer than just building the honeynet yourself. I am also going to strongly recommend not trying that. Realistically if you wanted to beautiful the dashboard then go for it but trying to rip the entire suite, then best of luck. Modern Honey Net might be something more up your alley as it will teach you how to integrate honeypots into a dashboard of your customization. Threatstream do a great job maintaining it however their deployment scripts certainly need some.looking over. Minor tweaks and changes.
> > @iukea1 I wouldn't use that git. Working on publishing some revisions for it shortly and it could get very funky.
> > https://github.com/threatstream/mhn
> 
> Well, even I was thinking that better to build the honeypots by myself rather than picking up the codes from there.
> 
> But from a license perspective I believe and feel its a open source product I guess.

hold the phone my bud. Open source is free for non-commercial use. (depending on the licensing that they set up with git)  

I would highly recommend you tread that water very carefully at your place of work, and get your legal guys to clear anything before going down this path.

**Comment by iukea1** [comment]
@Blason the only reason I am recommending that you don't truly go down this path is that I did it and it took months to do, and I ended up having to rewrite a lot of T-pots code bases and reverse engineer how they did everything... in the end 80% had to get rewritten and now there is not much of T-pot is left. Also if you are not good at docker and elk then this is not going to be fun.

Take it from a try hard.
